modules/host/negotiateformcontract.go

```diff
@@ -192,27 +182,27 @@ func (h *Host) managedFinalizeContract(builder modules.TransactionBuilder, rente
 // file contract, creating a storage obligation and submitting the contract to
 // the blockchain.
 func (h *Host) managedRPCFormContract(conn net.Conn) error {
-	// Set the negotiation deadline.
-	conn.SetDeadline(time.Now().Add(modules.NegotiateFileContractTime))
-
 	// Send the host settings to the renter. If the host is not accepting new
 	// contracts, the renter is expected to see this and gracefully handle the
 	// host closing the connection.
-	h.mu.RLock()
-	settings := h.settings
-	secretKey := h.secretKey
-	h.mu.RUnlock()
-	err := crypto.WriteSignedObject(conn, settings, secretKey)
+	err := h.managedRPCSettings(conn)
 	if err != nil {
 		return err
 	}
+
+	// If the host is not accepting contracts, the connection can be closed.
+	// The renter has been given enough information to understand that the
+	// connection is going to be closed.
+	h.mu.RLock()
+	settings := h.settings
+	h.mu.RUnlock()
 	if !settings.AcceptingContracts {
-		// The host is not accepting contracts, the connection can be closed.
-		// The renter has been given enough information to understand that the
-		// connection is going to be closed.
 		return nil
 	}
 
+	// Set the negotiation deadline.
+	conn.SetDeadline(time.Now().Add(modules.NegotiateFileContractTime))
```

Comment:
oh, I guess the settings adds its own deadline. got it.

So, I'm increasingly convinced that it would be beneficial for the host to recognize repeat renters. The biggest reason is repeat collateral, simply because collateral is such a sensitive resource. Mostly, I'm cautious about users who come in, try the system, and then leave, consuming a relatively large amount of host collateral without actually uploading many files. If we have 20x as many users as hosts, and each user is talking to a host, the host may find that they actually need a massive amount of coins to properly manage the collateral.

A tool such as renter recognition would go a long way. I'm not sure the best way to approach that, but we should keep that in mind moving forward.